packages/taco/integration-test/encrypt-decrypt.test.ts

```diff
@@ -189,5 +190,227 @@ describe.skipIf(!process.env.RUNNING_IN_CI)(
+
+      // Prepare for decryption
+      const messageKitFromBytes = ThresholdMessageKit.fromBytes(encryptedBytes);
+      const conditionContext =
+        conditions.context.ConditionContext.fromMessageKit(messageKitFromBytes);
+
+      // Verify that the context requires the ECDSA signature parameter
+      expect(conditionContext.requestedContextParameters.has(':ecdsaSignature')).toBe(true);
+
+      // Add the signature to the condition context in the format expected by Python
+      conditionContext.addCustomContextParameterValues({
+        ':ecdsaSignature': signatureHex
+      });
+
+      // Decrypt message
+      const decryptedBytes = await decrypt(
+        provider,
+        DOMAIN,
+        messageKitFromBytes,
+        conditionContext,
+      );
+      const decryptedMessageString = fromBytes(decryptedBytes);
+
+      // Verify decryption was successful
+      expect(decryptedMessageString).toEqual(messageString);
+
+      console.log('✅ ECDSA condition test passed!');
+      console.log(`   Decrypted Message: "${decryptedMessageString}"`);
+    }, 20000); // 20s timeout
+
+    test('should fail to decrypt with ECDSA condition when signature is invalid', async () => {
+      // Create test message
+      const messageString = 'This should fail with wrong signature';
+      const message = toBytes(messageString);
+
+      // Create a test ECDSA keypair
+      const ecdsaWallet = ethers.Wallet.createRandom();
+      const verifyingKey = ecdsaWallet.publicKey.slice(2);
+
+      // Message that needs to be signed
+      const authorizationMessage = 'I authorize access to this encrypted data';
+
+      // Create ECDSA condition
+      const ecdsaCondition = new conditions.base.ecdsa.ECDSACondition({
+        message: authorizationMessage,
+        signature: ':ecdsaSignature',
+        verifyingKey: verifyingKey,
+        curve: 'SECP256k1',
+      });
+
+      // Encrypt message
+      const messageKit = await encrypt(
+        provider,
+        DOMAIN,
+        message,
+        ecdsaCondition,
+        RITUAL_ID,
+        encryptorSigner,
+      );
+
+      const encryptedBytes = messageKit.toBytes();
+
+      // Prepare for decryption with INVALID signature
+      const messageKitFromBytes = ThresholdMessageKit.fromBytes(encryptedBytes);
+      const conditionContext =
+        conditions.context.ConditionContext.fromMessageKit(messageKitFromBytes);
+
+      // Add an INVALID signature (just random hex)
+      const invalidSignature = '0x' + randomBytes(64).toString('hex');
+      conditionContext.addCustomContextParameterValues({
+        ':ecdsaSignature': invalidSignature.slice(2)
+      });
+
+      // Attempt to decrypt should fail
+      await expect(
+        decrypt(
+          provider,
+          DOMAIN,
+          messageKitFromBytes,
+          conditionContext,
+        )
+      ).rejects.toThrow();
+
+      console.log('✅ ECDSA invalid signature test passed - decryption properly failed');
```

Comment:
nitpick: Let's skip on the console logs since they can be a lot. This is just the test passing. Applies to other locations in the file.